### PR TITLE
Chat QNA React UI, with conversation history

### PR DESCRIPTION
**Pull request by jaswanth8888**
## Description

Chat QNA UI using react 

## Issues

n/a

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies
 - mantine/core
 - mantine/hooks
 - microsoft/fetch-event-source
 - reduxjs/toolkit
 - tabler/icons-react
 - axios
 - react
 - react-dom
 - react-redux

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

File: ChatQnA/docker/ui/react/src/components/Conversation/Conversation.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import { getCurrentTimeStamp } from '../../common/util'
+import { useDisclosure } from '@mantine/hooks'
+import DataSource from './DataSource'
+import { ConversationSideBar } from './ConversationSideBar'
+
+type ConversationProps = {
+  title:string
+}
+
+const Conversation = ({ title }: ConversationProps) => {
```
Comment (yogeshmpandey) [inline ChatQnA/docker/ui/react/src/components/Conversation/Conversation.tsx:21]: Please fix the typo

File: ChatQnA/docker/ui/react/src/components/Conversation/ConversationSideBar.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export interface ConversationContextProps {
+    title: string
+}
+
+export function ConversationSideBar({ title }: ConversationContextProps) {
+    const { conversations, selectedConversationId } = useAppSelector(conversationSelector)
+    // const user = useAppSelector(userSelector)
+    const dispatch = useAppDispatch()
+
+    const converstaionList = conversations?.map((curr) => (
```
Comment (yogeshmpandey) [inline ChatQnA/docker/ui/react/src/components/Conversation/ConversationSideBar.tsx:20]: Please fix Typo in Variable name

File: ChatQnA/docker/xeon/README.md [hunk 3]
```diff
@@ -326,8 +340,24 @@ To access the frontend, open the following URL in your browser: http://{host_ip}
       - "80:5173"
 ```
 
+## 🚀 Launch the Conversational UI (react)
+
+To access the Conversational UI frontend, open the following URL in your browser: http://{host_ip}:5174. By default, the UI runs on port 80 internally. If you prefer to use a different host port to access the frontend, you can modify the port mapping in the `docker_compose.yaml` file as shown below:
+
+```yaml
+  chaqna-gaudi-ui-server:
```
Comment (yogeshmpandey) [inline ChatQnA/docker/xeon/README.md:348]: Application name to be updated to chaqna-xeon-conversation-ui-server

File: ChatQnA/docker/ui/react/src/components/Conversation/ConversationSideBar.tsx (hunk 2 above)
Comment (jaswanth8888) [reply ChatQnA/docker/ui/react/src/components/Conversation/ConversationSideBar.tsx]: fixed the typo

File: ChatQnA/tests/test_chatqna_on_xeon.sh [hunk 4]
```diff
@@ -57,6 +57,7 @@ function start_services() {
         sed -i "s#image: opea/chatqna:latest#image: opea/chatqna:${IMAGE_TAG}#g" docker_compose.yaml
         sed -i "s#image: opea/chatqna-ui:latest#image: opea/chatqna-ui:${IMAGE_TAG}#g" docker_compose.yaml
         sed -i "s#image: opea/*#image: ${IMAGE_REPO}opea/#g" docker_compose.yaml
+        sed -i "s#image: opea/chatqna-conversation-ui:latest#image: opea/chatqna-conversation-ui:${IMAGE_TAG}#g" docker_compose.yaml
```
Comment (chensuyue) [inline ChatQnA/tests/test_chatqna_on_xeon.sh:60]: This line should be after line 58.

Comment (jaswanth8888) [reply ChatQnA/tests/test_chatqna_on_xeon.sh:59]: Fixed

**Review by amberjain1 — APPROVED** [review]

**Comment by chensuyue** [comment]
@WenjiaoYue @lvliang-intel please review.

**Review by lvliang-intel — APPROVED** [review]

**Review by chensuyue — APPROVED** [review]

**Pull request merged**